reana-ui/src/pages/launchOnReana/LaunchOnReana.js

```diff
@@ -76,37 +87,107 @@ export default function LaunchOnReana() {
     return !requiredParams.every((param) => qsKeys.includes(param));
   }
 
+  /**
+   * Memoised function to get workflow parameters.
+   * Since this function is called twice and its return value should not
+   * change unless the query string parameters change, memoising it improves
+   * performace, avoiding extra calls.
+   */
+  const getWorkflowParameters = useCallback(() => {
+    /**
+     * Parses workflow parameters. Triggers notification error if invalid.
+     * @param {URLSearchParams} query Standard URLSearchParams object.
+     * @returns {Object | null} Returns object with workflow parameters or `null` if invalid.
+     */
+    function fn(query) {
+      try {
+        return JSON.parse(query.get("parameters"));
+      } catch (error) {
+        dispatch(
+          triggerNotification(
+            "An error has occurred",
+            `Invalid JSON workflow parameters provided: ${query.get(
+              "parameters"
+            )}`,
+            { error: true }
+          )
+        );
+      }
+      return null;
+    }
+    return fn(query);
+  }, [dispatch, query]);
+
   return (
     <BasePage>
       <Container text className={styles["container"]}>
         <div>
           <Title>Launch on REANA</Title>
         </div>
         {isMissingRequiredParams(query) ? (
-          <Message
-            icon="warning circle"
-            header="Missing required query string fields."
-            content={`The required fields: ${getRequiredQsParams()}`}
-            warning
-          />
+          <Welcome />
         ) : (
-          <ul>
-            {[...query].map(([k, v]) => (
-              <li key={k}>
-                {k}: {v}
-              </li>
-            ))}
-          </ul>
+          <Box className={styles.box} wrap>
+            <section className={styles.details}>
+              <span className={styles.icon}>
+                <Icon name="hourglass half" />
```

Comment:
Minor: I'm not sure about this icon. Since the workflow is not launched yet, it's kind of strange to see it as it gives the impression that something is loading already. WDYT?

Reply:
This was actually suggested by @tiborsimko [here](https://github.com/reanahub/reana-ui/issues/231#issuecomment-1065093382). Rocket didn't seem so good due to repetition and the cogwheel looks like settings to me. Do you have any other [suggestions](https://react.semantic-ui.com/elements/icon/)?